File: identhub/src/main/java/de/solarisbank/identhub/domain/verification/phone/AuthorizeVerificationPhoneUseCase.java
package de.solarisbank.identhub.domain.verification.phone;

import org.jetbrains.annotations.NotNull;

import de.solarisbank.identhub.data.verification.phone.model.VerificationPhoneResponse;
import de.solarisbank.identhub.domain.usecase.SingleUseCase;
import io.reactivex.Single;
import io.reactivex.android.schedulers.AndroidSchedulers;

public class AuthorizeVerificationPhoneUseCase extends SingleUseCase<Void, VerificationPhoneResponse> {
    private final VerificationPhoneRepository verificationPhoneRepository;

    public AuthorizeVerificationPhoneUseCase(VerificationPhoneRepository verificationPhoneRepository) {
        this.verificationPhoneRepository = verificationPhoneRepository;
    }

    @NotNull
    @Override
    protected Single<VerificationPhoneResponse> invoke(Void param) {
        return verificationPhoneRepository.authorize()
                .observeOn(AndroidSchedulers.mainThread());
    }
}

File: identhub/src/main/java/de/solarisbank/identhub/domain/verification/phone/AuthorizeVerificationPhoneUseCase.kt
package de.solarisbank.identhub.domain.verification.phone

import de.solarisbank.identhub.session.data.verification.phone.model.VerificationPhoneResponse
import de.solarisbank.sdk.data.entity.NavigationalResult
import de.solarisbank.sdk.domain.usecase.SingleUseCase
import io.reactivex.Single
import io.reactivex.android.schedulers.AndroidSchedulers

class AuthorizeVerificationPhoneUseCase(private val verificationPhoneRepository: VerificationPhoneRepository) : SingleUseCase<Unit, VerificationPhoneResponse>() {

    override fun invoke(param: Unit): Single<NavigationalResult<VerificationPhoneResponse>> {
        return verificationPhoneRepository.authorize()
                .map { NavigationalResult(it) }
                .observeOn(AndroidSchedulers.mainThread())
    }
}